_episodes_rmd/06_day5.Rmd

```diff
@@ -0,0 +1,260 @@
+
+> ## Solution
+> ```{r subset-vector-data}
+> 
+> WIcities_points$NAME
+> 
+> MM_points <- WIcities_points[WIcities_points$NAME=='Madison'|WIcities_points$NAME=='Milwaukee',]
+> ```
+{: .solution}
+
+# 3) Crop raster with vector 
+
+It may be useful to limit the raster data to only areas of interest. This is helpful when you load a larger raster file, such as data for the entire globe, but are only interested in a small part. For our capstone data, we loaded the global temperature data, but we are only interested in the data for Wisconsin, and specifically, the data for Milwaukee and Madison. Let's use the `crop()` function in the `raster` package to cut out the maximum temperature data to only the area for the Wisconsin cities.
+
+> ## Solution
+> ```{r crop-raster-by-vector}
+> 
+> maxTemp_WI <- crop(x = maxTemp_global, y = extent(WIcities_points),snap="out")
+> maxTemp_WI_df <- as.data.frame(maxTemp_WI, xy=T)
+> 
+> ggplot() +
+>   geom_raster(data = maxTemp_WI_df,aes(x = x, y = y, fill = bio5)) + 
+>   geom_sf(data = MM_points,color="yellow",size=5) + 
+>   coord_sf()
+> ```
+{: .solution}
+
+# 4) Summarize raster data by vector locations
+Now that we have our raster layer that contains the data we want to summarize, and the vector data that contains the locations indicating where we want to summarize the data, we can now put them together.
+
+If you are working with your own data, ask yourself:
+
++ How do I want to summarize the data? For example, do I want a single mean value? A maximum? A distribution of all values?
++ What is the area I want to summarize? If it's a point, do you want a certain distance around the point? Or, if it's an area, you will include all of the raster values within that area.
+
+## Extract pixel values
+
+For the capstone data, we want to extract all the pixel values within a distance buffer of the two city points. Since our data is "unprojected" and therefore the units are in latitude and longitude, the buffer distance will be in meters (hint: this information is in the Help file for the `extract` function!)
+
+> ## Geographic and projected unit reminder
+>
+> Remember, your raster is "unprojected" meaning that the pixels represent distances on a sphere of degrees, minutes, and seconds.
+> You'll need to consider is the converstion from degrees to meters. The conversions depends on where you are on the globe. For example, at the equator, 1 degree is 111 km. This distance decreases as you move closer to the poles.
+>
+{: .callout}
+
+> ## Solution
+> ```{r extract-raster-by-vector}
+> 
+> # extract pixel values from the raster based on the Milwaukee and Madison points
+> # using a buffer size of 50,000 (50 km around the each point) because the units are in meters
+> maxTemp_MM <- extract(x=maxTemp_global,df=TRUE,
+>                            y=MM_points,
+>                            buffer=50000)
```

Comment:
Should the prompt state to set the `buffer=50000` or are you allowing the learners to explore different buffers?